test/plugins/cxxrandom.lua

```diff
@@ -0,0 +1,65 @@
+local rng = require('plugins.cxxrandom')
+
+function test.cxxrandom_distributions()
+    rng.normal_distribution:new(0,5)
```

Comment:
I don't see this invocation style documented in the `cxxrandom` docs. Does this return an allocated object or an engine id? Can we test for the return value to be non-null? Does the returned object need to be destroyed/freed?

an end-to-end example of usage like this would be useful to have in the docs.

Reply:
it's a lua "object"
```lua
normal_distribution = {}
function normal_distribution:new(avg, stddev)
    local o = {}
    self.__index = self
    if type(avg) ~= 'number' or type(stddev) ~= 'number' then
        error("Invalid arguments in normal_distribution construction. Average and standard deviation must be numbers.")
    end
    o.average = avg
    o.std_deviation = stddev
    setmetatable(o,self)
    return o
end
```